rpcs3/Emu/Cell/lv2/sys_fs.cpp

```diff
@@ -209,60 +220,51 @@ u64 lv2_fs_mount_info_map::get_all(CellFsMountInfo* info, u64 len) const
 	return count;
 }
 
-std::string_view lv2_fs_object::get_device_root(std::string_view filename)
+bool lv2_fs_mount_info_map::vfs_unmount(std::string_view vpath)
 {
-	std::string_view mp_name, vpath = filename;
+	const std::string local_path = vfs::get(vpath);
 
-	for (usz depth = 0;;)
-	{
-		// Skip one or more '/'
-		const auto pos = vpath.find_first_not_of('/');
+	if (local_path.empty())
+		return false;
 
-		if (pos == 0)
+	if (fs::is_file(local_path))
+	{
+		if (fs::remove_file(local_path))
 		{
-			// No need to process it, return it as-is
-			return vpath;
+			sys_fs.notice("Removed simplefs file \"%s\"", local_path);
 		}
-
-		if (pos == umax)
+		else
 		{
-			break;
+			sys_fs.error("Failed to remove simplefs file \"%s\"", local_path);
 		}
+	}
 
-		// Get fragment name
-		const auto name = vpath.substr(pos, vpath.find_first_of('/', pos) - pos);
-		vpath.remove_prefix(name.size() + pos);
+	const auto result = vfs::unmount(vpath);
 
-		// Process special directories
-		if (name == "."sv)
-		{
-			// Keep current
-			continue;
-		}
+	if (result)
+		g_fxo->get<lv2_fs_mount_info_map>().remove(vpath);
```

Comment:
This code leads to a crash. You are removing from the map while iterating. The loop is on line 147.

Reply:
@oltolm can you try with #13961 ?

Reply:
I have, it doesn't crash anymore.

Reply:
Thank you two for working on that.
I feel sorry about that as the problems didn't occur when I was testing.